components/CourseCardGrid.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div class="course-card-grid">
+    <MediaCard
+      v-for="{
+        id,
+        category,
+        title,
+        hero_image,
+        slug,
+        units,
+        description,
+      } in courses"
+      :key="id"
+      :label="category"
+      :title="title"
+      :media="getCourseMediaUrl(hero_image)"
+      :card-link="getCardLink(slug, units)"
+    >
+      <template v-if="description" #body>
+        <p>
+          {{ description }}
+        </p>
+      </template>
+    </MediaCard>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import type { Course } from '@/types/course';
+import type { Unit } from '@/types/unit';
+
+defineProps<{
+  courses: Course[];
+}>();
+
+function getCourseMediaUrl(heroImage: string | undefined): string {
+  const runtimeConfig = useRuntimeConfig();
+  const { apiBase } = runtimeConfig;
+
+  // TODO: Change the placeholder with a default image
+  return heroImage
+    ? `${apiBase}/assets/${heroImage}`
+    : `/images/placeholder.png`;
+}
+
+function getCardLink(
+  courseSlug: string | undefined,
+  courseUnits: Unit[] | undefined
+): string | null {
```

Comment:
```suggestion
) {
```

We don't need to write the types, TypeScript can inherit it.

Reply:
Done with https://github.com/Qiskit/saiba/pull/421#discussion_r1057808497